src/admin_controlled.rs

```diff
@@ -0,0 +1,26 @@
+use crate::sdk;
+
+pub type PausedMask = u8;
```

Comment:
Might be worth using https://crates.io/crates/bitflags to make this (and constants `PAUSE_DEPOSIT` `PAUSE_WITHDRAW`) a little easier to read.

Reply:
Good suggestion, I like the approach of using `bitflags`!
Though we need to think if we change this right now because the current implementation is almost the same that we have for the `rainbow-bridge` and `rainbow-token-connector`.

Also I think that using this crate can bloat the contract size up. So we need to carefully decide about it.

Reply:
Sure, I don't have a strong opinion either way. If the crate isn't providing much value and we've implemented direct bitflags in other places then it's probably fine to leave it as is. I just thought it would mention it in case it was something you didn't know about.

Reply:
I'm pretty sure that all logic with `mask` should be refactored with `enum`.
```
pub enum ContractPause {
    Deposit = 0x0,
    Withdraw = 0x1,
}
```

Reply:
Bitflag is awesome and lightweight. Very very light library. I second that.